pkg/sql/plan2/base_binder.go

```diff
@@ -1138,3 +1151,34 @@ func resetDateFunctionArgs(dateExpr *Expr, intervalExpr *Expr) ([]*Expr, error)
 		},
 	}, nil
 }
+
+func resetDateFunctionArgs2(dateExpr *Expr, intervalExpr *Expr) ([]*Expr, error) {
+	switch intervalExpr.Expr.(type) {
+	case *plan.Expr_List:
+		return resetDateFunctionArgs(dateExpr, intervalExpr)
+	}
+	list := &plan.ExprList{}
+	list.List = append(list.List, intervalExpr)
```

Comment:
little improve:
list := &plan.ExprList{
   List : make([]*Expr, 2),
}
this case we always get two item in list

Reply:
ok, I will change it